biorbd_optim/optimal_control_program.py

```diff
@@ -459,11 +461,137 @@ def add_plot(self, fig_name, update_function, phase_number=-1, **parameters):
 
         nlp["plot"][plot_name] = custom_plot
 
+    def export_eocar_ode_model(self):
+        from acados_template import AcadosModel
+        from casadi import MX, Function, external, SX, vertcat, sin, cos
+
+
+        m = biorbd.Model(self.nlp[0]['model'].path().relativePath().to_string())
+        model_name = "model_name"
+
+        # Declare model variables
+        x = MX.sym('x', m.nbQ() * 2)
+        u = MX.sym('u', m.nbQ())
+        xdot = MX.sym('dx', m.nbQ() * 2)
+
+        f = Function('f', [x, u],
+                     [vertcat(x[m.nbQ():], m.ForwardDynamics(x[:m.nbQ()], x[m.nbQ():], u).to_mx())]).expand()
+        x = SX.sym('x', m.nbQ() * 2)
+        u = SX.sym('u', m.nbQ())
+        xdot = SX.sym('dx', m.nbQ() * 2)
+        f_expl = f(x, u)
+        f_impl = xdot - f_expl
+
+        model = AcadosModel()
+        model.f_impl_expr = f_impl
+        model.f_expl_expr = f_expl
+        model.x = x
+        model.xdot = xdot
+        model.u = u
+        # model.z = z
+        model.p = []
+        model.name = model_name
+
+        return model
+
+    def __prepare_acados(self):
+        from acados_template import AcadosOcp, AcadosOcpSolver, AcadosModel
+        import numpy as np
+        import scipy.linalg
+        from casadi import Function, external, SX, sin, cos
+
+        os.environ["ACADOS_SOURCE_DIR"] = "/home/dangzilla/Documents/Programmation/acados"
+        # create ocp object to formulate the OCP
+        ocp = AcadosOcp()
+
+        # # set model
+        model = self.export_eocar_ode_model()
+        ocp.model = model
+
+        # set time
+        ocp.solver_options.tf = self.nlp[0]["tf"]
+
+        # set dimensions
+        ocp.dims.nx = self.nlp[0]["nx"]
```

Comment:
Please use "i" with i = 0 instead of 0 for specific phase related index so we just have to implement the loop (and not have to retrieve where 0 were hardcoded

Reply:
I think limiting it to 1 phase for now is the right approach. Make sure you raise a NotImplementedYet error if self.nb_phases is not one though so we keep in mind that it only works for 1 phase for now